collect_app/src/test/java/org/odk/collect/android/activities/SplashScreenActivityTest.kt

```diff
@@ -67,56 +64,10 @@ class SplashScreenActivityTest {
         })
     }
 
-    @Test
-    fun `SplashScreenActivity should implement ProjectAddedListener`() {
-        doReturn(true).`when`(splashScreenViewModel).shouldFirstLaunchScreenBeDisplayed
-
-        val scenario = ActivityScenario.launch(SplashScreenActivity::class.java)
-        scenario.onActivity { activity: SplashScreenActivity ->
-            assertThat(activity is ProjectAddedListener, `is`(true))
-        }
-    }
-
-    @Test
-    fun `The main menu should be displayed when onProjectAdded() is called`() {
-        doReturn(true).`when`(splashScreenViewModel).shouldFirstLaunchScreenBeDisplayed
-
-        val scenario = ActivityScenario.launch(SplashScreenActivity::class.java)
-        scenario.onActivity { activity: SplashScreenActivity ->
-            Intents.init()
-            activity.onProjectAdded()
-            Intents.intended(hasComponent(MainMenuActivity::class.java.name))
-            Intents.release()
-        }
-    }
-
-    @Test
-    fun `The First Launch Screen should be displayed if the app is newly installed`() {
-        doReturn(true).`when`(splashScreenViewModel).shouldFirstLaunchScreenBeDisplayed
-
-        val scenario = ActivityScenario.launch(SplashScreenActivity::class.java)
-        scenario.onActivity { activity: SplashScreenActivity ->
-            assertThat(RobolectricHelpers.getFragmentByClass(activity.supportFragmentManager, FirstLaunchDialog::class.java), `is`(notNullValue()))
-        }
-    }
-
-    @Test
-    fun `The First Launch Screen should not be displayed if the app is not newly installed`() {
-        doReturn(false).`when`(splashScreenViewModel).shouldFirstLaunchScreenBeDisplayed
-        doReturn(true).`when`(splashScreenViewModel).shouldDisplaySplashScreen
-        doReturn(false).`when`(splashScreenViewModel).doesLogoFileExist
-
-        val scenario = ActivityScenario.launch(SplashScreenActivity::class.java)
-        scenario.onActivity { activity: SplashScreenActivity ->
-            assertThat(RobolectricHelpers.getFragmentByClass(activity.supportFragmentManager, FirstLaunchDialog::class.java), `is`(nullValue()))
-        }
-    }
-
     @Test
     fun `The Splash screen should be displayed with the default logo if it's enabled and no other logo is set`() {
-        doReturn(false).`when`(splashScreenViewModel).shouldFirstLaunchScreenBeDisplayed
-        doReturn(true).`when`(splashScreenViewModel).shouldDisplaySplashScreen
-        doReturn(false).`when`(splashScreenViewModel).doesLogoFileExist
+        `when`(splashScreenViewModel.shouldDisplaySplashScreen).thenReturn(true)
+        `when`(splashScreenViewModel.doesLogoFileExist).thenReturn(false)
```

Comment:
I changed all these to use when/thenReturn which feels more readable to me. `doReturn` is typically only recommended if necessary: https://www.oreilly.com/library/view/test-driven-development-with/9781783283293/ch06s10.html

Reply:
Sure. 👍